### PR TITLE
TT-10675 add SQS Pump Backend support

### PR DESCRIPTION
**Pull request by mativm02**
This PR is a copy of the Pull Request submitted by [masoudhaghbin](https://github.com/masoudhaghbin) - https://github.com/TykTechnologies/tyk-pump/pull/740

Add SQS pump to supported backends
## Description

The PR will add the support to pump logs into a SQS instance

## Related Issue

https://github.com/TykTechnologies/tyk-pump/issues/737

## Types of changes

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [x] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [x] I have updated the documentation accordingly.
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/774/merge
Commit:  
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk-pump/actions/runs/7625156199)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/774/merge
Commit:  
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk-pump/actions/runs/7625156199)

**Review by tbuchaillot — APPROVED** [review]

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: bf9e7e71760f8cb8fbf2243402eb3123d434543f TT-10675 add SQS Pump Backend support (#774)

* add SQS Pump Backend support

* add conditions for groupID

* add conditions for delay seconds

* add deduplication for FIFO Queues

* add sqs backend to readme

* add comments for sqs configs

* remove pretty print and use log.debug

* add sqs tests

* linting

---------

Co-authored-by: Masoud Haghbin <masoud.haghbin7@gmail.com>
Co-authored-by: Masoud Haghbin <masoud.haghbin@company.info>
Triggered by: push (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk-pump/actions/runs/7639541708)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: bf9e7e71760f8cb8fbf2243402eb3123d434543f TT-10675 add SQS Pump Backend support (#774)

* add SQS Pump Backend support

* add conditions for groupID

* add conditions for delay seconds

* add deduplication for FIFO Queues

* add sqs backend to readme

* add comments for sqs configs

* remove pretty print and use log.debug

* add sqs tests

* linting

---------

Co-authored-by: Masoud Haghbin <masoud.haghbin7@gmail.com>
Co-authored-by: Masoud Haghbin <masoud.haghbin@company.info>
Triggered by: push (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk-pump/actions/runs/7639541708)

File: README.md [hunk 1]
```diff
@@ -1268,6 +1269,76 @@ TYK_PMP_PUMPS_CSV_TYPE=csv
...
+
+`aws_message_group_id` - Specifies the name of the AWS Simple Queue Service (SQS) queue where messages will be sent
+
+`aws_sqs_batch_limit` - Sets the maximum number of messages to include in a single batch when sending messages to the SQS queue
+
+`aws_message_id_deduplication_enabled` - Enables or disables the deduplication of messages based on unique message IDs to prevent unintended duplicates in the queu
+
+`aws_delay_seconds` - Configures the delay (in seconds) before messages sent to the SQS queue become available for processing.
+
+When you initialize a Timestream Pump, the SDK uses its default credential chain to find AWS credentials. This default credential chain looks for credentials in the following order:
```
Comment (kolavcic) [inline README.md:1290]: @mativm02 I believe we have a typo here - it should say When you initialize a SQS Pump

**Pull request merged**